CSIP5: Create testCase.xml

koit opened the pull request.

koit committed:
Create testCase.xml for CSIP5

koit committed:
Shorten corpus package names and descriptions

PhillipTommerholt commented on corpora/csip/metadata/mets-root/CSIP5/testCase.xml:
```diff
@@ -1,23 +1,88 @@
-  <id specification="E-ARK CSIP" version="2.0-DRAFT" requirementId="CSIP5"/>
-  <!-- URL references to requirements for convenient lookup -->
+<testCase xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="testCase.xsd" testable="TRUE">
+  <id specification="CSIP" version="2.0.3" requirementId="CSIP5"/>
```
2.0.4 is the latest

koit replied:
Fixed.

PhillipTommerholt replied:
It says 2.0.3 and not 2.0.4?

koit replied:
I think I get the issue now: GitHub keeps linking to the version that was valid at the moment of initial review, so the updates made by the later commit 2875094fc9c609c7eed60194ee9f715eec595635 are not there. It seems the "Files changed" tab of the PR always displays the latest state.

PhillipTommerholt commented on corpora/csip/metadata/mets-root/CSIP5/testCase.xml:
```diff
@@ -1,23 +1,88 @@
+      When the mets/@csip:CONTENTINFORMATIONTYPE has the value "OTHER" the mets/@csip:OTHERCONTENTINFORMATIONTYPE must state the content information type.
+    </description>
+    <cardinality>0..1</cardinality>
+    <!--  It is actually a MUST or more precisely, a "conditional MUST":
```
Yes - I think we need to discuss this with A2. How to represent conditional MUSTs in all of our specifications?

koit replied:
I've created a ticket about it: https://github.com/DILCISBoard/E-ARK-CSIP/issues/657.

PhillipTommerholt commented on corpora/csip/metadata/mets-root/CSIP5/testCase.xml:
```diff
@@ -1,23 +1,88 @@
+    </rule>
+    <!-- This rule is implicit in the specification and is important for CSIP4 and CSIP5 to work as intended. -->
+    <rule id="2">
+      <description>If mets/@csip:OTHERCONTENTINFORMATIONTYPE exists then mets/@csip:CONTENTINFORMATIONTYPE MUST be "OTHER".</description>
```
Or maybe also "MIXED"?

koit replied:
I see the concern but I don't have enough experience to know if it's an issue in real life. If it is, we should propose a change to the A2 team. At the moment the CSIP5 description mentions only "OTHER," so I left it unchanged.

PhillipTommerholt commented on corpora/csip/metadata/mets-root/CSIP5/testCase.xml:
```diff
@@ -1,23 +1,88 @@
+      <description>If mets/@csip:OTHERCONTENTINFORMATIONTYPE exists then mets/@csip:CONTENTINFORMATIONTYPE MUST be "OTHER".</description>
+      <error level="ERROR">
+        <message>
+          mets/@csip:OTHERCONTENTINFORMATIONTYPE exists although mets/@csip:CONTENTINFORMATIONTYPE is not "OTHER". MUST be: mets/@csip:OTHERCONTENTINFORMATIONTYPE MUST only exist if mets/@csip:CONTENTINFORMATIONTYPE = "OTHER".
```
Or maybe also "MIXED"?

PhillipTommerholt reviewed (changes requested):
I think there might lack a test to make sure that the value in @othercontentinformationtype is not empty.
Maybe it can come with the list of the values that must not exist in ruleID=1:
- vocabulary items
- empty string

koit committed:
Update testCase.xml

Move rules 1 and 2 from CSIP4 to CSIP5.
Shorten package names.
Add paths.
Fix spec version.

koit commented:
> I think there might lack a test to make sure that the value in @othercontentinformationtype is not empty.
> Maybe it can come with the list of the values that must not exist in ruleID=1:
> 
> * vocabulary items
> * empty string

Fixed. The rules for emptiness existed but were part of CSIP4 testCase - I've now moved them here.

PhillipTommerholt reviewed (changes requested):
It says 2.0.3 and not 2.0.4 in line 2?

carlwilson committed:
Merge branch 'integration' into csip/mets-root/csip5
